tests_func/helpers/mongo.go

```diff
@@ -0,0 +1,306 @@
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"github.com/docker/docker/api/types"
+	u "github.com/wal-g/wal-g/tests_func/utils"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func connectHostPort(context context.Context, host string, port uint16) *mongo.Client {
+	uri := fmt.Sprintf("mongodb://%s:%d/?connect=direct", host, port)
+	client, err := mongo.NewClient(options.Client().ApplyURI(uri))
+	if err != nil {
+		panic(err)
+	}
+	err = client.Connect(context)
+	if err != nil {
+		panic(err)
+	}
+	return client
+}
+
+func connect(context context.Context, user string, password string, dbname string, host string, port uint16) *mongo.Client {
+	uri := fmt.Sprintf("mongodb://%s:%s@%s:%d/%s?connect=direct&authMechanism=SCRAM-SHA-1", user, password, host, port, dbname)
+	client, err := mongo.NewClient(options.Client().ApplyURI(uri))
+	if err != nil {
+		panic(err)
+	}
+	err = client.Connect(context)
+	if err != nil {
+		panic(err)
+	}
+	return client
+}
+
+func EnvDBConnect(testContext *TestContextType, nodeName string) *mongo.Client {
+	dbMongoPort, err := strconv.Atoi(u.GetVarFromEnvList(testContext.Env, "MONGO_EXPOSE_MONGOD"))
+	if err != nil {
+		panic(err)
+	}
+	dbHost := GetDockerContainer(testContext, nodeName)
+	host, port := GetExposedPort(*dbHost, uint16(dbMongoPort))
+	conn := connectHostPort(
+		testContext.Context,
+		host,
+		port)
+	return conn
+}
+
+func EnvDBConnectWithCreds(testContext *TestContextType, nodeName string, creds UserConfiguration) *mongo.Client {
+	dbMongoPort, err := strconv.Atoi(u.GetVarFromEnvList(testContext.Env, "MONGO_EXPOSE_MONGOD"))
+	if err != nil {
+		panic(err)
+	}
+	dbHost := GetDockerContainer(testContext, nodeName)
+	host, port := GetExposedPort(*dbHost, uint16(dbMongoPort))
+	conn := connect(
+		testContext.Context,
+		creds.Username,
+		creds.Password,
+		creds.Dbname,
+		host,
+		port)
+	return conn
+}
+
+func FillWithData(context context.Context, database *mongo.Client, mark string) map[string]map[string][]DatabaseRecord {
+	var data = make(map[string]map[string][]DatabaseRecord, 0)
+	for i := 1; i <= 2; i++ {
+		dbName := fmt.Sprintf("test_db_%02d", i)
+		if _, ok := data[dbName]; !ok {
+			data[dbName] = map[string][]DatabaseRecord{}
+		}
+		for j := 1; j <= 2; j++ {
+			var rows []DatabaseRecord
+			var irows []interface{}
+			tableName := fmt.Sprintf("test_table_%02d", j)
+			for k := 1; k <= 2; k++ {
+				rows = append(rows, generateRecord(k, 5, mark))
+				irows = append(irows, generateRecord(k, 5, mark))
+			}
+			_, err := database.Database(dbName).Collection(tableName).InsertMany(context, irows)
+			if err != nil {
+				panic(err)
+			}
+			data[dbName][tableName] = rows
+		}
+	}
+	return data
+}
+
+type DatabaseRecord struct {
+	Datetime time.Time
+	IntNum   int
+	Str      string
+}
+
+func generateRecord(rowNum int, strLen int, strPrefix string) DatabaseRecord {
```

Comment:
can we use something like https://github.com/Percona-Lab/sysbench-mongodb-lua here somehow?